apm-sniffer/apm-sdk-plugin/grpc-1.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/grpc/v1/server/TracingServerCallListener.java

```diff
@@ -76,15 +76,19 @@ public void onCancel() {
         final AbstractSpan span = ContextManager.createLocalSpan(operationPrefix + REQUEST_ON_CANCEL_OPERATION_NAME);
         span.setComponent(ComponentsDefine.GRPC);
         span.setLayer(SpanLayer.RPC_FRAMEWORK);
-        ContextManager.continued(contextSnapshot);
+        if (contextSnapshot != null) {
```

Comment:
If no contextSnapshot to continue, are we still reasonable to create a local span? I think this would be a no-parent span.

Reply:
@wu-sheng I just have verified it. you are right. if the `onCancel` creates local span, it is just alone.
<img width="1138" alt="image" src="https://user-images.githubusercontent.com/29882506/161978904-5dc79b21-dafb-4d48-8e67-514b58c33b4d.png">

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/29882506/161978956-11a97aa9-285d-4cf0-af1d-570d5bcddf9f.png">

this trace data is useless. if the `contextSnapshot` is null, the `onCancel` method should directly return.  as for `asyncSpan`, it don't need to be added not-null condition. because if the `contextSnapshot` is not null, it proved that the `onReady` method is executed. Then the  value  of `asyncSpan` must not be null.